src/java.base/share/classes/java/io/FileSystem.java

```diff
@@ -243,18 +243,21 @@ public abstract boolean createFileExclusively(String pathname)
      */
     public abstract int hashCode(File f);
 
-    // Flags for enabling/disabling performance optimizations for file
-    // name canonicalization
-    static final boolean useCanonCaches;
-    static final boolean useCanonPrefixCache;
 
     private static boolean getBooleanProperty(String prop, boolean defaultVal) {
         String value = System.getProperty(prop);
         return (value != null) ? Boolean.parseBoolean(value) : defaultVal;
     }
 
     static {
-        useCanonCaches      = getBooleanProperty("sun.io.useCanonCaches", false);
-        useCanonPrefixCache = useCanonCaches && getBooleanProperty("sun.io.useCanonPrefixCache", false);
+        infoPropIsIgnored("sun.io.useCanonCaches");
```

Comment:
I am not sure we should warn. If we should, there might be a better way to do it?